### ISSUE TITLE
deprecated: delete envoy_stream_options

### ISSUE THREAD

**Issue by junr03** [lyft/envoy-mobile#643]
stream options was added in order to configure buffering for retries. This was required when the direct API leveraged the AsyncClient. However, after #616 the option is no longer needed as the Router uses the configured retry policy in the route/and or headers in the stream to determine if retries are enabled and thus if body data has to be buffered.

After #616 merges envoy_stream_options can be removed. Note, for Lyft, that this is a breaking API change so the Apps function calls will have to be updated when the parameter is removed.

**Comment by rebello95** [comment]
https://github.com/lyft/envoy-mobile/pull/689
